app/main.py

```diff
@@ -2,0 +1,36 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}"
+                f", Health: {self.health}"
+                f", Hidden: {self.hidden}}}")
+
+    @classmethod
+    def remove_dead(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+    @classmethod
+    def __str__(cls) -> str:
+        return str([repr(animal) for animal in cls.alive])
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.health = 0
+                Animal.remove_dead()
```

Comment:
Calling `Animal.remove_dead()` here might be redundant since the health is already set to zero. Consider whether this call is necessary or if it can be handled elsewhere.